File: step1/break_repeating-key_xor/edit_distance.c
#include "../../cryptoutils.h"


int main()
{
	char string1[15] = {"this is a test"};
	char string2[15] = {"wokka wokka!!!"};

	int editdistance = get_edit_distance(strlen(string1),string1,strlen(string2),string2);
	printf("%i\n",editdistance);

	return 0;
}

File: step1/break_repeating-key_xor/edit_distance.py
def string2bits(string):
    bits = list()
    for x in string:
        bits.append(format(ord(x),"08b"))

    return bits

string1 = "this is a test"
string2 = "wokka wokka!!!"

list1 = string2bits(string1)
list2 = string2bits(string2)

diferences = 0
for x in range(len(list1)):
    for y in range(8):
        if list1[x][y] != list2[x][y]:
            diferences += 1

print diferences

#print list1
#print list2
